docs/developer/architecture/snmp.md

```diff
@@ -103,6 +106,199 @@ snmp_listener:
 
 ### Cluster Agent Support
 
-For Kubernetes environments, the [Cluster Agent](https://docs.datadoghq.com/agent/cluster_agent/) can be configured to use the Agent auto-discovery logic as a source of [Cluster checks](https://docs.datadoghq.com/agent/cluster_agent/clusterchecks/).
+For Kubernetes environments, the [Cluster Agent](https://docs.datadoghq.com/agent/cluster_agent/) can be configured to use the SNMP Agent auto-discovery (uses snmp listener) logic as a source of [Cluster checks](https://docs.datadoghq.com/agent/cluster_agent/clusterchecks/).
+
+![](../../assets/images/snmp-architecture-cluster-agent.png)
+
+The Datadog Cluster Agent (DCA) uses the `snmp_listener` config (Agent auto-discovery) to listen for IP ranges, then schedule snmp check instances to be run by one or more normal Datadog Agents.
+
+Agent auto-discovery combined with Cluster Agent is very scalable, it can be used to monitor a large number of snmp devices.
```

Comment:
```suggestion
Node/Daemonset agent auto-discovery combined with Cluster Agent is very scalable, it can be used to monitor a large number of snmp devices.
```

Can we clarify that because I think of the Host agent when refering to single `Agent`

Reply:
`Agent auto-discovery` is the `snmp_listener` feature, they are a bit used interchangeably.

In this doc, `snmp_listener` is run by the Cluster Agent.

Here when talking about `Cluster Agent`, I mean `Cluster Agent` with the node agents (kind of workers).